crates/bevy_core_pipeline/src/taa/taa.wgsl

```diff
@@ -81,50 +74,45 @@ fn taa(@location(0) uv: vec2<f32>) -> Output {
     let texture_size = vec2<f32>(textureDimensions(view_target));
     let texel_size = 1.0 / texture_size;
 
-    // Fetch the current sample
-    let original_color = textureSample(view_target, nearest_sampler, uv);
-    var current_color = original_color.rgb;
-#ifdef TONEMAP
-    current_color = tonemap(current_color);
-#endif
-
-#ifndef RESET
-    // Pick the closest motion_vector from 5 samples (reduces aliasing on the edges of moving entities)
-    // https://advances.realtimerendering.com/s2014/index.html#_HIGH-QUALITY_TEMPORAL_SUPERSAMPLING, slide 27
-    let offset = texel_size * 2.0;
-    let d_uv_tl = uv + vec2(-offset.x, offset.y);
-    let d_uv_tr = uv + vec2(offset.x, offset.y);
-    let d_uv_bl = uv + vec2(-offset.x, -offset.y);
-    let d_uv_br = uv + vec2(offset.x, -offset.y);
+    // Loop over 3x3 neighborhood of the pre-TAA rendered texture
+    // https://alextardif.com/TAA.html
+    var current_color = vec3(0.0);
+    var weight_sum = 0.0;
+    var moment_1 = vec3(0.0);
+    var moment_2 = vec3(0.0);
+    var closest_depth = 0.0;
     var closest_uv = uv;
-    let d_tl = textureSample(depth, nearest_sampler, d_uv_tl);
-    let d_tr = textureSample(depth, nearest_sampler, d_uv_tr);
-    var closest_depth = textureSample(depth, nearest_sampler, uv);
-    let d_bl = textureSample(depth, nearest_sampler, d_uv_bl);
-    let d_br = textureSample(depth, nearest_sampler, d_uv_br);
-    if d_tl > closest_depth {
-        closest_uv = d_uv_tl;
-        closest_depth = d_tl;
-    }
-    if d_tr > closest_depth {
-        closest_uv = d_uv_tr;
-        closest_depth = d_tr;
-    }
-    if d_bl > closest_depth {
-        closest_uv = d_uv_bl;
-        closest_depth = d_bl;
-    }
-    if d_br > closest_depth {
-        closest_uv = d_uv_br;
+    var weights = array(1.5894572e-7, 0.8888889, 1.5894572e-7);
+    for (var x = -1.0; x <= 1.0; x += 1.0) {
+        for (var y = -1.0; y <= 1.0; y += 1.0) {
+            let sample_uv = uv + (vec2(x, y) * texel_size);
+            let sample = sample_view_target(sample_uv);
+
+            // Apply Mitchell-Netravali kernel over the jittered 3x3 neighborhood to reduce softness
+            let weight = weights[u32(x + 1.0)] * weights[u32(y + 1.0)];
+            weight_sum += weight;
+            current_color += sample * weight;
+
+            // Calculate first and second color moments for use with variance clipping
+            moment_1 += sample;
+            moment_2 += sample * sample;
+
+            // Find closest pixel to take motion vectors from (reduces aliasing on the edges of moving entities)
+            let sample_depth = textureSampleLevel(depth, nearest_sampler, sample_uv, 0.0);
+            if sample_depth > closest_depth {
+                closest_depth = sample_depth;
+                closest_uv = sample_uv;
+            }
+        }
     }
-    let closest_motion_vector = textureSample(motion_vectors, nearest_sampler, closest_uv).rg;
+    current_color /= weight_sum;
```

Comment:
weight_sum can be calculated as a constant here.

Reply:
Good point

Reply:
now the weights sum to precisely 1 so you can actually remove this